test/typescript/custom-types/t.test.ts

```diff
@@ -174,18 +179,61 @@ describe('t', () => {
       expectTypeOf(t('dessert', { context: ctx })).toMatchTypeOf<string>();
     });
 
-    it('should trow error with string union with missing context value', () => {
+    it('should throw error with string union with missing context value', () => {
       enum DessertMissingValue {
-        COOKIE = 'cookie',
         CAKE = 'cake',
         MUFFIN = 'muffin',
         ANOTHER = 'another',
       }
 
-      const ctxMissingValue = DessertMissingValue.ANOTHER;
+      const getRandomDessert = (): DessertMissingValue =>
+        Math.random() < 0.5 ? DessertMissingValue.CAKE : DessertMissingValue.ANOTHER;
+
+      const ctxRandomValue: DessertMissingValue = getRandomDessert();
+
+      // @ts-expect-error Dessert.ANOTHER is not mapped so it must give a type error
+      expectTypeOf(t('dessert', { context: ctxRandomValue })).toMatchTypeOf<string>();
 
       // @ts-expect-error Dessert.ANOTHER is not mapped so it must give a type error
-      expectTypeOf(t('dessert', { context: ctxMissingValue })).toMatchTypeOf<string>();
+      expectTypeOf(t('dessert', { context: DessertMissingValue.ANOTHER })).toMatchTypeOf<string>();
+
+      expectTypeOf(
+        // @ts-expect-error 'another' is not mapped so it must give a type error
+        t('dessert', { context: 'cake' as 'cake' | 'another' }),
+      ).toEqualTypeOf<'a nice cake'>();
+
+      // TODO: edge case which is not correctly detected currently
+      // expectTypeOf(
+      //   // @ts-expect-error no default context so it must give a type error
+      //   t('dessert', { context: 'cake' as 'cake' | undefined }),
+      // ).toEqualTypeOf<never>();
```

Comment:
@marcalexiei while using this in my actual app I found this edge case which is uncovered. Wasn't able to figure out how to correctly handle this yet, maybe you have an idea. Overall I think it's not a blocker for the PR as it's still an improvement over the current functionality, but if possible to find a solution for this as well it would be great.

Reply:
also can in theory be covered by writing code like this `value ? t('dessert', { context: value }) : t('dessert')`. Which is not ideal but helps.